direct/common/subsample.py

```diff
@@ -1714,33 +2186,621 @@ def mask_func(
+        crop_corner : bool, optional
+            If True, the mask is cropped to the corners. Default: True.
+        """
+        super().__init__(
+            accelerations=accelerations,
+            center_fractions=center_fractions,
+            uniform_range=uniform_range,
+        )
+        self.crop_corner = crop_corner
+
+    def mask_func(
+        self,
+        shape: Union[list[int], tuple[int, ...]],
+        return_acs: bool = False,
+        seed: Optional[Union[int, Iterable[int]]] = None,
+    ) -> torch.Tensor:
+        """Creates a kt radial mask.
+
+        Parameters
+        ----------
+        shape : list or tuple of ints
+            The shape of the mask to be created. The shape should at least 3 dimensions.
+            Samples are drawn along the second last dimension.
+        return_acs : bool
+            Return the autocalibration signal region as a mask.
+        seed : int or iterable of ints or None (optional)
+            Seed for the random number generator. Setting the seed ensures the same mask is generated
+             each time for the same shape. Default: None.
+
+        Returns
+        -------
+        torch.Tensor
+            The sampling mask.
+        """
+        if len(shape) not in [4, 5]:
+            raise ValueError("Shape should have 4 or 5 dimensions.")
+
+        (nt, num_rows, num_cols) = shape[-4:-1]
+
+        with temp_seed(self.rng, seed):
+
+            center_fraction, acceleration = self.choose_acceleration()
+            num_low_freqs = int(round(num_cols * center_fraction))
+
+            offset_angle = self.rng.uniform(0, 360)
+
+        acs_mask = self.zero_pad_to_center(np.ones((nt, num_low_freqs, num_low_freqs)), [nt, num_rows, num_cols])
+
+        if return_acs:
+            return torch.from_numpy(acs_mask.astype(bool)[np.newaxis, ..., np.newaxis])
+
+        adjusted_acceleration = (acceleration * (num_low_freqs**2 - num_rows * num_cols)) / (
+            num_low_freqs**2 * acceleration - num_rows * num_cols
+        )
+
+        rate = 1 / adjusted_acceleration
+        beams = int(rate * np.mean([num_rows, num_cols]))  # beams is the number of angles
+
+        if self.crop_corner:
+            temp_size = max(num_rows, num_cols)
+        else:
+            temp_size = int(np.sqrt(num_rows**2 + num_cols**2))
+
+        aux = np.zeros((temp_size, temp_size))
+        aux[int(temp_size / 2), :] = 1
+
+        mask = np.zeros((nt, num_rows, num_cols))
+        for i in range(nt):
+            angles = np.linspace(0 + offset_angle * i, 180 + offset_angle * i + 1, beams)
+            mask_t = np.zeros((num_rows, num_cols))
+            for ang in angles:
+                temp = self.crop_center(rotate(aux, ang, reshape=False, order=0), num_rows, num_cols)
+                mask_t += temp
+            mask[i] = mask_t
+
+        mask = mask + acs_mask
+        mask = mask > 0
+
```

Comment:
is there no way to do this in a comprehension? If you can do it in pytorch you can also use torchscript to speed this up significantly (use jit)

Reply:
This is done more efficiently now (no loop) - and it's pretty fast.